jax/interpreters/pxla.py

```diff
@@ -345,30 +320,56 @@ def block_until_ready(self):
   @property
   def _value(self):
     if self._npy_value is None:
-      ids = self._ids()
+      # TODO(skye): remove this to avoid transferring replicated buffers?
```

Comment:
Are you asking about the `ids = self._ids()` line? It was added specifically to avoid extra copies to host.

I think the analogue now would be to look at `self.logical_indices` and figure out a plan of what minimal set of indices cover the logical array. Forming that plan optimally [might be NP-hard](https://en.wikipedia.org/wiki/Set_cover_problem) if indices could be arbitrary subsets (like integer array indexing), but since we've constrained them to be intervals maybe there's something better.

Reply:
The below line, `self.copy_to_host_async()`. This seems unnecessary and possibly incurring extra copies (in both the current code and this PR), but maybe I'm missing something.

Reply:
@hawkinsp, do you remember what `self.copy_to_host_async()` is doing here?

Reply:
It's triggering the transfers for all of the device buffers before blocking on the transfer for any of them.